ha/monitor/k8s/object_monitor.py

```diff
@@ -167,3 +172,44 @@ def run(self):
                 break
         Log.info(f"Stopping the {self.name}...")
 
+    def _is_published_alert(self, alert) -> bool:
+        """
+        Check incoming alert is already published or not.
+        If incoming alert is not found in published alerts then,
+        it is a new alert to publish.
+        self._published_alerts = { resource_name: [alert1, alert2] }
+
+        Returns:
+            True if it is published already
+            False if it is a new alert
+        """
+        incoming_alert = alert.to_dict().copy()
+
+        # Alert with no status change also has new timestamp.
+        # So timestamp field should be ignored for validation.
+        if not incoming_alert.get('_timestamp') is not None:
+            return False
+        del incoming_alert['_timestamp']
+
+        # Remove user added fields those aren't exist in raw event
+        if incoming_alert.get('_is_status') is not None:
+            del incoming_alert['_is_status']
+
+        resource_name = incoming_alert.get('_resource_name')
+
+        if resource_name:
+            incoming_alert_msg = json.dumps(incoming_alert, sort_keys=True)
+            published_alerts = self._published_alerts.get(resource_name)
+            if published_alerts and isinstance(published_alerts, list):
+                if incoming_alert_msg in published_alerts:
+                    return True
+                else:
+                    published_alerts.append(incoming_alert_msg)
```

Comment:
We need to store only one alert per resource_name right ?  Why do we need a list ?

Reply:
During pod deletion, current pod is getting terminated and new pod is getting created.
So there are two pods events one by one per resource_name. That's why list is used.

If I don't use list,
1. New pod creation event will be caught initially (published - online)
2. Pod which is getting deleted will overwrite that (published -failed)
This would still send that new pod (online) event which will become duplicate of (1).

Reply:
There is an another scenario I need to cover. I will fulfil the gap in next commit.

Reply:
@ArchanaLimaye can you check now.

Reply:
Done.